plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/CacheMediatorTransformer.java

```diff
@@ -77,19 +79,36 @@ public void transformWithinSequence(TransformationInfo information,
 		setCommonProperties(cacheMediator, visualCache);
 		{
 			if (visualCache.getCacheAction().getValue() == 0) {
-				cacheMediator.setHTTPMethodsToCache(visualCache.getCacheProtocolMethods().split(","));
 				cacheMediator.setProtocolType(visualCache.getCacheProtocolType().getLiteral());
-				cacheMediator.setResponseCodes(visualCache.getResponseCodes());
+				if (CachingConstants.HTTP_PROTOCOL_TYPE.equals(visualCache.getCacheProtocolType().getLiteral())) {
+					if (StringUtils.isNotBlank(visualCache.getCacheProtocolMethods())) {
```

Comment:
Shall we format here? Spacing between line 83 and 84, and 84 and 85 differs.

Reply:
Formatting issues are fixed in : #272